### PR TITLE
Save Recipe action implementation

### PR DESCRIPTION
**Pull request by xrsula**
This PR deals with the linking between the save button of a recipe and the profile of the user saving the recipe. There are several things implemented in this issue:

- implement the saved button action in order to save a recipe in the profile view model
- display in the saved recipes screen the recipes that the user saved
- implement the saved button action in order to unsave a recipe in the profile view model
- edit the bookmark picture to make sure it is filled in when recipe is saved, and not filled in when not saved

### CODE REVIEW

**Comment by xrsula** [comment]
In this version, the backend is mostly implemented. The only issue is to find a way to refresh the landing page screen every time we click on the bookmark to save something, since the bookmark stays blank even though the recipe was added to the saved recipes list and screen

**Comment by xrsula** [comment]
I had to create a new recipe since the test for the saved button action failed when the list of recipes in the landing page was a duplicate of the same recipe (which makes sense since the landing page will not display duplicate recipes)

**Review by JeannedeM — COMMENTED** [review]
Hey, nice job !! It's working really well :D Everything looks good to me except this small bug : 
image
When you click here on th bookmark to save the recipe, nothing happens :(.
See also my small comment on one function

File: app/src/main/java/com/android/feedme/model/viewmodel/ProfileViewModel.kt [hunk 1]
```diff
@@ -414,4 +417,24 @@ class ProfileViewModel : ViewModel() {
...
+
+  /**
+   * Removes a saved recipe from the current user's saved recipes.
+   *
+   * @param recipe The recipe ID to remove from the saved recipes.
+   */
+  fun removeSavedRecipes(recipes: List<Recipe>) {
+    _currentUserSavedRecipes.value = _currentUserSavedRecipes.value.filter { it !in recipes }
+  }
+
```
[the remark this answers is not in the document]
Comment (xrsula) [reply app/src/main/java/com/android/feedme/model/viewmodel/ProfileViewModel.kt:438]: Yess I will fix all of that ASAP :)

File: app/src/main/java/com/android/feedme/model/viewmodel/ProfileViewModel.kt [hunk 2]
```diff
@@ -414,4 +417,24 @@ class ProfileViewModel : ViewModel() {
...
         uri = picture)
   }
+
+  /**
+   * Adds a saved recipe to the current user's saved recipes.
+   *
+   * @param recipes The list of recipe IDs to add to the saved recipes.
+   */
+  fun addSavedRecipes(recipes: List<Recipe>) {
+    _currentUserSavedRecipes.value += recipes
```
[the remark this answers is not in the document]
Comment (xrsula) [reply app/src/main/java/com/android/feedme/model/viewmodel/ProfileViewModel.kt:427]: Ah yes you're right ! I will fix this ASAP :)

**Comment by ohamz** [comment]
Went from 34% to 54% line coverage but it's getting late...

**Comment by xrsula** [comment]
> Went from 34% to 54% line coverage but it's getting late...

Thanks for your help 🥹 I will pick it up from here !

**Review by fdejean — APPROVED** [review]
Great back-end implementation !
Thank you for the effort you put into quickly implementing it, as it is clearly not an easy task :)

**Pull request merged**